common/file.c

```diff
@@ -374,38 +374,30 @@ ods_replace(const char *str, const char *oldstr, const char *newstr)
         return buffer;
     }
 
+    // Length of str upto the first occurence of oldstr
     part1_len = ch-str;
+
+    // Length of newstr (which will be inserted after part 1)
     part2_len = strlen(newstr);
-    part3_len = strlen(ch+strlen(oldstr));
+
+    // Length of the string that follows oldstr
+    char* part3 = ch+strlen(oldstr);
+    part3_len = strlen(part3);
+
     buffer = calloc(part1_len+part2_len+part3_len+1, sizeof(char));
     if (!buffer) {
         return NULL;
     }
+    buffer[0] = '\0';
 
-    if (part1_len) {
-        strncpy(buffer, str, part1_len);
-        buffer[part1_len] = '\0';
-
-        if (part2_len) {
-            strncat(buffer, str, part2_len);
-            buffer[part1_len+part2_len] = '\0';
-        }
-    } else {
-        strncpy(buffer, newstr, part2_len);
-        buffer[part2_len] = '\0';
-    }
-
-    if (part3_len) {
-        strncat(buffer, ch+strlen(oldstr), part3_len);
-        buffer[part1_len+part2_len+part3_len] = '\0';
-    }
+    // Characters preceeding oldstr exist in str. Copy them to the output buffer.
+    strncat(buffer, str, part1_len);
+    strcat(buffer, newstr);
+    strcat(buffer, part3);
```

Comment:
```suggestion
    strncat(buffer, newstr, part2_len);
    strncat(buffer, part3, part3_len);
```
`strcat` is considered unsafe or dangerous. If you ask me `strncat` is just as unsafe to be honest, but general opinion is that it is "better" anyway.

Reply:
Ah, this was changed to silence warnings from GCC.

Reply:
With your proposed changes GCC complains:

```
../file.c: In function ‘ods_replace’:
../file.c:395:5: warning: ‘strncat’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
  395 |     strncat(buffer, newstr, part2_len);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../file.c:381:17: note: length computed here
  381 |     part2_len = strlen(newstr);
      |                 ^~~~~~~~~~~~~~
../file.c:396:5: warning: ‘strncat’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
  396 |     strncat(buffer, part3, part3_len);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../file.c:385:17: note: length computed here
  385 |     part3_len = strlen(part3);
      |                 ^~~~~~~~~~~~~
```

Reply:
Okay... I'll adapt my suggestion to use `strlcat` :)